frontend/src/pages/Dashboard/index.tsx

```diff
@@ -1,48 +1,120 @@
+import { useEffect, useState } from "react";
 import "./styles.css";
 import UserGroupCard from "@/components/UserGroupCard";
 import { Table, TableRow, TableCell } from "@/components/Table/index.tsx";
 import { Link } from "react-router-dom";
 
+interface IAssignment {
+    id: number; 
+    rubric_id: number | null; 
+    active: boolean;
+    assignment_classroom_id: number;
+    semester_id: number;
+    name: string;
+    local_id: number;
+    main_due_date: Date | null;
+  }
+
 const Dashboard: React.FC = () => {
+    const [assignments, setAssignments] = useState<IAssignment[]>([]);
+    const options: Intl.DateTimeFormatOptions = {
+        weekday: 'short', year: 'numeric', month: 'short', day: 'numeric',
+        hour: '2-digit', minute: '2-digit', timeZoneName: 'short'
+      };
+
+    // API call when the component is rendered
+    useEffect(() => {
+        const fetchAssignments = async () => {
+            try {
+                const base_url: string = import.meta.env.VITE_PUBLIC_API_DOMAIN as string;
+                const result = await fetch(`${base_url}/assignments`, {
+                    method: 'GET',
+                    credentials: 'include',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                });
+
+                if (!result.ok) {
+                    throw new Error('Network response was not ok');
+                }
+
+                const data: IAssignment[] = (await result.json() as IAssignment[])
+                console.log(data)
+                const assignmentGoodDate = data.map((assignment: IAssignment) => ({
+                    ...assignment,
+                    main_due_date: assignment.main_due_date ? new Date(assignment.main_due_date) : null,
+                }))
+                setAssignments(assignmentGoodDate); 
+            } catch (error) {
+                console.error('Error fetching assignments:', error);
+            } finally {
+                console.log("Successful Fetch")
+            }
+        };
+
+        const SyncWithClassroom = async () => {
+            try {
+                const base_url: string = import.meta.env.VITE_PUBLIC_API_DOMAIN as string;
+                const result = await fetch(`${base_url}/github/sync`, {
+                    method: 'POST',
+                    credentials: 'include',
+                    headers: {
+                        'Content-Type': 'application/json',
+                    },
+                    body: JSON.stringify({ classroom_id: 237210 }), //237209 // TODO: UnHardcode when we have this info programatically
```

Comment:
commenting here to remind you of hard-coded value

Reply:
done